### PR TITLE
Han Support https://hanzi.pro/ #1558 #1222 #698

### PR DESCRIPTION
**Pull request by geekrainy**

### CODE REVIEW

File: layout/_partials/head.swig [hunk 1]
```diff
@@ -3,6 +3,11 @@
 <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
 
 
+{% if theme.han %}
+  <link rel="stylesheet" media="all" href="//cdnjs.cloudflare.com/ajax/libs/Han/3.2.7/han.min.css">
```
Comment (Acris) [inline layout/_partials/head.swig]: cloudflare的CDN在国内很慢吧？
建议不要直接引用CDN，直接将资源文件放在主题里，同时提供CDN选项，可以参考其他资源文件。

**Comment by iissnan** [comment]
Thanks. :+1:

File: layout/_macro/post.swig [hunk 2]
```diff
@@ -249,7 +249,7 @@
     {% endif %}
 
 
-    <div class="post-body" itemprop="articleBody">
+    <div class="post-body han-init-context" itemprop="articleBody">
```
Comment (ivan-nginx) [inline layout/_macro/post.swig]: Where is the switch for `han-init-context`?

Comment (geekrainy) [inline layout/_macro/post.swig]: It's just a hook. When Han is not enabled, it will not have any effect.

Comment (ivan-nginx) [inline layout/_macro/post.swig]: But it's will be in default class style.
<div class="post-body{% if theme.han %} han-init-context{% endif %}" itemprop="articleBody">

Comment (geekrainy) [inline layout/_macro/post.swig]: Thanks for your suggestion. 756c49cc05680ba512060c4d3046e1a3cf218377

Comment (ivan-nginx) [inline layout/_macro/post.swig]: In ideal, this switches must be anywhere on any custom options.

**Comment by ivan-nginx** [comment]
References to #1558 #1222 #698.

**Pull request merged**